deltacat/compute/compactor/steps/repartition.py

```diff
@@ -0,0 +1,162 @@
+    repartition_args: dict,
+    destination_partition: Partition,
+    enable_profiler: bool,
+    read_kwargs_provider: Optional[ReadKwargsProvider],
+    repartitioned_file_content_type: ContentType = ContentType.PARQUET,
+    deltacat_storage=unimplemented_deltacat_storage,
+) -> RepartitionResult:
+
+    if repartition_type == RepartitionType.RANGE:
+        column: str = repartition_args["column"]
+        partition_ranges: List = repartition_args["ranges"]
+        task_id = get_current_ray_task_id()
+        worker_id = get_current_ray_worker_id()
+        with memray.Tracker(
+            f"repartition_{worker_id}_{task_id}.bin"
+        ) if enable_profiler else nullcontext():
+            tables: List[pa.Table] = deltacat_storage.download_delta(
+                annotated_delta,
+                storage_type=StorageType.LOCAL,
+                file_reader_kwargs_provider=read_kwargs_provider,
+            )
+            # check if the column exists in the table
+            if not all(column in table.column_names for table in tables):
+                raise ValueError(f"Column {column} does not exist in the table")
+            # given a range [x, y, z], we need to split the table into 4 files, i.e., (-inf, x], (x, y], (y, z], (z, inf)
+            partitioned_tables_list = [[] for _ in range(len(partition_ranges) + 1)]
+            total_record_count = 0
+            col_name_int64 = f"{column}_int64"
+            for table in tables:
+                total_record_count += len(table)
+                table_new = table.add_column(
+                    0,
+                    pa.field(col_name_int64, pa.int64()),
+                    pc.cast(table[column], pa.int64()),
+                )
+                # handle the partition for values less than or equal to the smallest value
+                partitioned_tables_list[0].append(
+                    table_new.filter(
+                        pc.field(col_name_int64) <= pc.scalar(partition_ranges[0])
+                    )
+                )
+                # Iterate over pairs of values in partition_ranges
+                for i, (lower_limit, upper_limit) in enumerate(
+                    zip(partition_ranges[:-1], partition_ranges[1:]), start=1
+                ):
+                    # Add the table filtered by the lower and upper limits to partitioned_tables
+                    partitioned_tables_list[i].append(
+                        table_new.filter(
+                            (pc.field(col_name_int64) > pc.scalar(lower_limit))
+                            & (pc.field(col_name_int64) <= pc.scalar(upper_limit))
+                        )
+                    )
+                # handle the partition for values greater than the largest value
+                partitioned_tables_list[-1].append(
+                    table_new.filter(
+                        pc.field(col_name_int64) > pc.scalar(partition_ranges[-1])
+                    )
+                )
```

Comment:
Can these statements be avoided by add an -Inf and +Inf in the list of partition ranges?